src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,465 @@
+                     result( res )
+    ! Uniformly distributed integer in [loc, loc + scale]
+    !
+        ${t1}$, intent(in) :: loc, scale
+        ${t1}$  ::  res
+
+        if(scale <= 0_${k1}$) call error_stop("Error(unif_dist_rvs): Uniform"   &
+                         //" distribution scale parameter must be positive")
+        res = loc + unif_dist_rvs_1_${t1[0]}$${k1}$(scale)
+    end function unif_dist_rvs_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function unif_dist_rvs_0_${t1[0]}$${k1}$( ) result(res)
+    ! Uniformly distributed float in [0,1]
+    ! Based on the paper by Frederic Goualard, "Generating Random Floating-
+    ! Point Numbers By Dividing Integers: a Case Study", Proceedings of
+    ! ICCS 2020, June 2020, Amsterdam, Netherlands
+    !
+        ${t1}$  ::  res
+        integer(int64) :: tmp
+
+        tmp = shiftr(dist_rand(INT_ONE), 11)        ! Get random from [0,2^53-1]
+        res = real(tmp * MESENNE_NUMBER, kind =${k1}$) ! convert to [0,1]
+    end function unif_dist_rvs_0_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function unif_dist_rvs_1_${t1[0]}$${k1}$(scale) result(res)
+    ! Uniformly distributed float in [0, scale]
+    !
+        ${t1}$, intent(in) :: scale
+        ${t1}$  ::  res
+
+        if(scale == 0._${k1}$) call error_stop("Error(unif_dist_rvs_1): "       &
+                      //"Uniform distribution scale parameter must be non-zero")
+        res = scale * unif_dist_rvs_0_${t1[0]}$${k1}$( )
+    end function unif_dist_rvs_1_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental function unif_dist_rvs_${t1[0]}$${k1}$(loc, scale)         &
+                     result(res)
+    ! Uniformly distributed float in [loc, loc + scale]
+    !
+        ${t1}$, intent(in) :: loc, scale
+        ${t1}$  ::  res
+
+        if(scale == 0._${k1}$) call error_stop("Error(unif_dist_rvs): "         &
+                      //"Uniform distribution scale parameter must be non-zero")
+        res = loc + scale * unif_dist_rvs_0_${t1[0]}$${k1}$( )
+    end function unif_dist_rvs_${t1[0]}$${k1}$
+
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    impure elemental function unif_dist_rvs_1_${t1[0]}$${k1}$(scale) result(res)
+    ! Uniformly distributed complex in [(0,0i), (scale, i(scale)]
+    ! The real part and imaginary part are independent of each other, so that
+    ! the joint distribution is on an unit square [(0,0i), scale,i(scale)]
+    !
+        ${t1}$, intent(in) :: scale
+        ${t1}$ :: res
+        real(${k1}$) :: r1, r2, tr, ti
+
+        if(scale == (0.0_${k1}$, 0.0_${k1}$)) call error_stop("Error(uni_dist_" &
+              //"rvs_1): Uniform distribution scale parameter must be non-zero")
+        r1 = unif_dist_rvs_0_r${k1}$( )
+        if(real(scale) == 0.0_${k1}$) then
+            ti = aimag(scale) * r1
```

Comment:
why not usinf the Fortran 2008 feature:
```suggestion
        if(scale % re == 0.0_${k1}$) then
            ti = scale % im * r1
```

Reply:
The reason for not using Fortran 2008 feature is because the new feature was not recognized 3 months ago during compilation on gfortran 7. If we no longer compile code on gfortran 7, it is certainly desirable to use new complex number feature.

Reply:
OK. GFortran 7 is not supported anymore (see [here](https://github.com/fortran-lang/stdlib#supported-compilers) for more details). The Fortran 2008 feature is also used in `stdlib` (e.g. [here](https://github.com/fortran-lang/stdlib/blob/ce3a106f14afcc01b8d2e48437c94e7c5e7a19a1/src/stdlib_math_linspace.fypp#L71))

Reply:
Changes were made for new feature.